editor/scene_tree_dock.cpp

```diff
@@ -1459,21 +1459,26 @@ void SceneTreeDock::_create() {
 			ERR_FAIL_COND(!parent);
 		}
 
+		// user selects node type to create
 		Object *c = create_dialog->instance_selected();
 
 		ERR_FAIL_COND(!c);
 		Node *child = Object::cast_to<Node>(c);
 		ERR_FAIL_COND(!child);
 
+		// begin recording "Create Node" do/undo action
```

Comment:
Most comments you added are not particularly useful, as they just say exactly what the code says already. I understand that as a new user of the Godot code base it might make sense to document this, but regular contributors see such calls dozens of time a day while reading the code base, so it actually adds a lot of unnecessary noise.

Comments are of course welcome in the code base, but they should explain things that the code doesn't already make obvious. Here `editor_data->get_undo_redo().create_action(TTR("Create Node"));` is quite self explanatory, reading from left to right, the UndoRedo class of EditorData is fetched, and a new action is created with the translatable name "Create Node".

General documentation of how UndoRedo works can be useful to have, but it should be in the developer documentation or in the UndoRedo C++ file itself, not copied in the hundreds of occurrences of UndoRedo all over the code base.

Same thing for:
```
// abort if nothing selected
ERR_FAIL_COND(selection.size() <= 0);
```
`ERR_FAIL_COND` is used 4545 times in the code base, and that's why it was given an explicit name: "ERRor if FAILing CONDition". Adding such a comment brings no value to the reader, unless it's the reader's first ever encounter with Godot code (but we can't make all source code a tutorial for new users, as existing users also need to be able to be productive with it).

So I would remove most of the comments you added personally.

Reply:
I see your point. Going to remove probably all my comments when I have time this week, as they are mostly a side-effect of me following the logic to see how the change node action was NOT getting recorded. Will update the PR soon.

Reply:
That said, we do see the need to have more developer-focused documentation for people getting into the codebase, we're just unsure what form it should take. I'll open an issue about that for further discussion.

Reply:
@akien-mga I believe I have incorporated the changes you suggested, though I'm not sure if all trailing spaces have been eliminated. If not it may be an encoding issue on my side that I need to further look into.

I am interested in helping with introductory dev documentation as I explore the codebase myself.